pulsar/impl_partition_producer.go

```diff
@@ -134,7 +131,7 @@ func (p *partitionProducer) grabCnx() error {
 	})
 
 	if err != nil {
-		p.log.WithError(err).Error("Failed to create producer")
+		log.Errorf("create producer [%s] error: %+v", *p.producerName, err)
```

Comment:
If you remove the context, all these logs will have no sense. The logs are supposed to be relative to some event which will have some context, eg. `topic=my-topic, producer-name=my-producer`. These are vital fro the logs. 

Repeating the the context information each time has 2 problems: 
 * It's hard to be consistent throughout the code base in terms of naming and parameters
 * We'd be losing the semi-structure of the logs

Saving the logger as part of the state ensure that we always have the context, added in a consistent way and will let the application process the logs easily.

Reply:
Yes, the context information is the meaning of the log output. Users can quickly locate the problem based on this information. Here, I did not remove the context information, but changed the way to output.

Reply:
In the above example, the topic name was already set in the `p.log` context and got missing in the refactoring.